lib/memfs.rb

```diff
@@ -100,6 +100,37 @@ def deactivate!
   end
   module_function :deactivate!
 
+  # Switches back to the original file system, calls the given block (if any),
+  # and switches back afterwards.
+  # 
+  # Allows for file & dir operations that need to take place on the real fs.
+  # 
+  # @example
+  #   MemFs.halt do
+  #     puts Dir.getwd
+  #   end
+  #   
+  # @return nothing
+  def halt
+
+    switch_to = -> dir_class, file_class { 
```

Comment:
Ups, sorry for the many violations - feel free to rewrite the code if you like the feature. I'm still a sort of apprentice when it comes to Ruby...